src/main/java/edu/byu/cs/autograder/QualityAnalyzer.java

```diff
@@ -64,82 +66,68 @@ private QualityAnalysis parseAnalysis(String output) {
                 warnings.add(trimmed);
             }
         }
-        return new QualityAnalysis(errors, warnings);
     }
 
-    private float getScore(QualityAnalysis analysis) {
+    private float getScore() {
         float totalPoints = 0;
         float earnedPoints = 0;
         for (QualityRubricCategory category : qualityRubricItems.categories()) {
             totalPoints += category.value();
-            earnedPoints += getCategoryPoints(analysis, category);
+            if(categoryPasses(category)) earnedPoints += category.value();
         }
         return earnedPoints / totalPoints;
     }
 
-    private String getResults(QualityAnalysis analysis) {
+    private String getResults() {
         StringBuilder resultsBuilder = new StringBuilder();
         for (QualityRubricCategory category : qualityRubricItems.categories()) {
             StringBuilder categoryResultsBuilder = new StringBuilder();
-            for (QualityRubricItem item : category.items()) {
-                for (QualityRubricReporter reporter : item.reporters()) {
-                    StringBuilder reporterResultsBuilder = new StringBuilder();
-                    if (analysis.errors().containsKey(reporter.name())) {
-                        analysis.errors().get(reporter.name())
-                                .forEach(s -> reporterResultsBuilder.append("\t\t").append(s).append("\n"));
-                    }
-                    if (!reporterResultsBuilder.isEmpty()) {
-                        categoryResultsBuilder.append("\t").append(reporter.name()).append(":\n")
-                                .append(reporterResultsBuilder);
-                    }
+            for(String reporter : category.reporters()) {
+                StringBuilder reporterResultsBuilder = new StringBuilder();
+                if (errors.containsKey(reporter)) {
+                    errors.get(reporter).forEach(s -> reporterResultsBuilder.append("\t\t").append(s).append("\n"));
```

Comment:
I'm sure this code _works_ just perfectly, and I can tell that it's doing some kind of string concatenation to build a resulting string that will be displayed to the user. 

It's kind of hard for me to read what kind of string this code is supposed to be building. Sometimes comments are problematic because they grow stale, but I wonder if it would be helpful here to have an example string of the output so I can follow along in the code...

Reply:
Example output:
Code Decomposition:
	DuplicateBlock:
		[ERROR] /server/src/main/java/service/Service.java:19:34: Duplicated code from /server/src/main/java/server/Server.java:170 [DuplicateBlock]
Code Readability:
	UnusedMethodReporter:
		[ERROR] /shared/src/main/java/model/GameData.java:30: Unused method clearBoard [UnusedMethodReporter]
		[ERROR] /client/src/main/java/ui/ChessClient.java:75: Unused method quit [UnusedMethodReporter]
		[ERROR] /client/src/main/java/ui/ChessClient.java:179: Unused method legal [UnusedMethodReporter]
		[ERROR] /client/src/main/java/ui/ChessClient.java:122: Unused method list [UnusedMethodReporter]
		[ERROR] /client/src/main/java/ui/ChessClient.java:170: Unused method redraw [UnusedMethodReporter]

Reply:
That's helpful to have in the PR. I suppose anyone working on the code will just need to reference the working product. It's probably safer to do it that way then risk having stale comments in the codebase.

Looks good to me